sympy/integrals/integrals.py

```diff
@@ -93,6 +93,14 @@ def __new__(cls, function, *symbols, **assumptions):
     def __getnewargs__(self):
         return (self.function,) + tuple([tuple(xab) for xab in self.limits])
 
+    # fixes issue #20809
+    def has_free(obj, syms):
+        bs = []
+        for _ in recur_free_in(obj, syms, bs):
```

Comment:
Creating the list should be handled in `recur_free_in` (which could use a nested function for the recursion part).

Reply:
`(which could use a nested function for the recursion part)`
Sorry, but I didn't get this idea. 
Kindly elaborate on it, please.
Thanks.

Reply:
can you just make the signature contain `bs` instantiation? : `recur_free_in(obj, syms, bs=[])`?